server/src/main/java/org/elasticsearch/action/admin/indices/datastream/DeleteDataStreamAction.java

```diff
@@ -176,16 +200,6 @@ static ClusterState removeDataStream(MetadataDeleteIndexService deleteIndexServi
                 snapshottingDataStreams.addAll(SnapshotsService.snapshottingDataStreams(currentState, dataStreams));
             }
 
-            if (dataStreams.isEmpty()) {
-                // if only a match-all pattern was specified and no data streams were found because none exist, do not
-                // fail with data stream missing exception
-                if (request.names.length == 1 && Regex.isMatchAllPattern(request.names[0])) {
-                    return currentState;
-                }
-                throw new ResourceNotFoundException("data_streams matching [" + Strings.arrayToCommaDelimitedString(request.names) +
-                    "] not found");
-            }
-
```

Comment:
This is a significant change to the behavior of the delete data stream request when a specified data stream is not present, but without this change, lots of tests running with security enabled fail because security resolves wildcard patterns to individual names including indices and those incorrectly trigger this `ResourceNotFoundException` error.

Reply:
I also don't see a way around this.